File: Difficulty: Medium/Indexes of Subarray Sum/indexes-of-subarray-sum.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    // Function to find a continuous sub-array which adds up to a given number.
    vector<int> subarraySum(vector<int> arr, int n, long long s) {
        // Your code here
        // 01.07.24 potd
        long long int sum = arr[0];
        int r = 0, l = 0;
        
        while(r < n){
            if(sum == s){
                return {l+1, r+1};
            } else if(sum < s){
                r++;
                sum += arr[r];
            } else if(sum > s && l == r){
                l++, r++;
                sum = arr[l];
            } else {
                sum -= arr[l];
                l++;
            }
        }
        
        
        return {-1};
    }
};

//{ Driver Code Starts.

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        long long s;
        cin >> n >> s;
        vector<int> arr(n);
        // int arr[n];
        const int mx = 1e9;
        for (int i = 0; i < n; i++) {
            cin >> arr[i];
        }
        Solution ob;
        vector<int> res;
        res = ob.subarraySum(arr, n, s);

        for (int i = 0; i < res.size(); i++)
            cout << res[i] << " ";
        cout << endl;
    }
    return 0;
}
// } Driver Code Ends

File: Difficulty: Medium/Indexes of Subarray Sum/indexes-of-subarray-sum.py
#User function Template for python3
class Solution:
    def subarraySum(self, arr, target):
        # code here
        # 09.01.25 POTD
        sumPos = {}
        sm, n = 0, len(arr)
        
        for i in range(n):
            sm += arr[i]
            
            if sm == target:
                return [1,i+1]
                
            if (sm - target) in sumPos:
                return [sumPos[sm - target] + 1, i + 1]
                
            sumPos[sm] = i + 1
            
        return [-1]

#{ 
 # Driver Code Starts
# Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input().strip())
    while tc > 0:
        arr = list(map(int, input().strip().split()))
        d = int(input().strip())
        ob = Solution()
        result = ob.subarraySum(arr, d)
        print(" ".join(map(str, result)))
        tc -= 1
        print("~")

# } Driver Code Ends